plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -2161,13 +2235,389 @@ fu_genesys_usbhub_device_update_firmware(FuGenesysUsbhubDevice *self,
 	fu_progress_step_done(progress);
 
 	/* verify */
+	if (!self->has_hw_codesign) { /* skip specific codesign verification */
+		if (!fu_genesys_usbhub_device_compare_flash_data(self,
+								 start_addr,
+								 buf,
+								 bufsz,
+								 fu_progress_get_child(progress),
+								 error))
+			return FALSE;
+	}
+	fu_progress_step_done(progress);
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_genesys_usbhub_device_setup_hw_module(FuGenesysUsbhubDevice *self, GError **error)
+{
+	GUsbDevice *usb_device = fu_usb_device_get_dev(FU_USB_DEVICE(self));
+
+	if (!g_usb_device_control_transfer(usb_device,
+					   G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+					   G_USB_DEVICE_REQUEST_TYPE_VENDOR,
+					   G_USB_DEVICE_RECIPIENT_DEVICE,
+					   GENESYS_USBHUB_GL_HUB_HW_SECURITY,
+					   0,	 /* value */
+					   0,	 /* idx */
+					   NULL, /* data */
+					   0,	 /* data length */
+					   NULL, /* actual length */
+					   GENESYS_USBHUB_USB_TIMEOUT,
+					   NULL,
+					   error)) {
+		g_prefix_error(error, "error setuping HW module: ");
```

Comment:
not sure setuping is a word -- although I do like it :) maybe `setting up` instead?